commit 13449f7dc2ac0d205180611f2de6e1371cd88a4b
Author: Zeeshan Lakhani <zeeshan.lakhani@gmail.com>
Date:   Thu May 7 00:20:08 2015 +0900

    Update patches for Solr 4.10.4

diff --git a/solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java b/solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java
index f574f29..42168df 100644
--- a/solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java
+++ b/solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java
@@ -29,6 +29,7 @@ import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
 import org.apache.commons.lang.StringUtils;
+import org.apache.http.params.HttpConnectionParams;
 import org.apache.http.client.HttpClient;
 import org.apache.http.impl.client.DefaultHttpClient;
 import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
@@ -208,5 +207,7 @@ public class HttpShardHandlerFactory extends ShardHandlerFactory implements org.
     httpRequestExecutor = new InstrumentedHttpRequestExecutor(this.metricNameStrategy);
     clientConnectionManager = new InstrumentedPoolingHttpClientConnectionManager(HttpClientUtil.getSchemaRegisteryProvider().getSchemaRegistry());
     this.defaultClient = HttpClientUtil.createClient(clientParams, clientConnectionManager, false, httpRequestExecutor);
+    this.defaultClient.getParams().setParameter(HttpConnectionParams.STALE_CONNECTION_CHECK, false);
+    this.defaultClient.getParams().setParameter(HttpConnectionParams.TCP_NODELAY, true);
     this.loadbalancer = createLoadbalancer(defaultClient);
     }
